src/CSET/operators/convection.py

```diff
@@ -0,0 +1,149 @@
+
+    .. math:: 1 - (\frac{SBCAPE}{MUCAPE})
+
+    The ratio is defined in this way such that if SBCAPE=MUCAPE the ratio will
+    equal 1. If the ratio was reversed when MUCAPE exists and SBCAPE is zero the
+    ratio would be undefined.
+
+    The diagnostic varies smoothly between zero and unity. A value of 0 implies
+    an environment is suitable for surface-based convection. A value of 1
+    implies an environment is suitable for elevated convection. Values between
+    imply transition convection with values closer to one imply elevated
+    convection is more likely and values closer to zero implying that
+    surface-based convection is more likely.
+
+    Further details about this diagnostic for elevated convection identification
+    can be found in Flack et al. (2023) [2]_.
+
+    Expected applicability ranges: Convective-scale models will be noisier than
+    parametrized models as they are more responsive to the convection, and thus
+    it may be more sensible to view as a larger spatial average rather than
+    grid point.
+
+    Interpretation notes: UM stash for CAPE and CIN are calculated at the end of
+    the timestep. Therefore this diagnostic is applicable after precipitation has
+    occurred, not before as is the usual interpretation of CAPE related diagnostics.
+
+    You might encounter ``RuntimeWarning: divide by zero encountered in divide``
+    or ``RuntimeWarning: invalid value encountered in divide`` this is expected
+    for when CAPE is zero. The data will be replaced by NaNs.
+
+
+
+    References
+    ----------
+    .. [1] Clark, A. J., Kain J. S., Marsh P. T., Correia J., Xue M., and Kong
+       F., (2012) "Forecasting tornado pathlengths using a three-dimensional
+       object identification algorithm applied to convection-allowing
+       forecasts." Weather and Forecasting, vol. 27, 1090–1113, doi:
+       10.1175/WAF-D-11-00147.1
+    .. [2] Flack, D.L.A., Lehnert, M., Lean, H.W., and Willington, S. (2023)
+       "Characteristics of Diagnostics for Identifying Elevated
+       Convection over the British Isles in a Convection-Allowing Model."
+       Weather and Forecasting, vol. 30, 1079-1094, doi:
+       10.1175/WAF-D-22-0219.1
+
+    Examples
+    --------
+    >>> CAPE_ratios_filter=elevated_convection_diagnostics.cape_ratio(
+            SBCAPE,MUCAPE,MUCIN)
+    >>> iplt.pcolormesh(CAPE_ratios[0,:,:],cmap=mpl.cm.RdBu)
+    >>> plt.gca().coastlines('10m')
+    >>> plt.colorbar()
+    >>> plt.clim(0,1)
+    >>> plt.show()
+
+    >>> CAPE_ratios_filter=elevated_convection_diagnostics.cape_ratio(
+            SBCAPE,MUCAPE,MUCIN,MUCIN_thresh=-1.5)
+    >>> iplt.pcolormesh(CAPE_ratios_filter[0,:,:],cmap=mpl.cm.RdBu)
+    >>> plt.gca().coastlines('10m')
+    >>> plt.clim(0,1)
+    >>> plt.colorbar()
+    >>> plt.show()
+    """
+    # Setup new arrays so do not overwrite cubes.
+    SBCAPE_data = np.zeros(np.shape(SBCAPE.data))
+    MUCAPE_data = np.zeros(np.shape(MUCAPE.data))
```

Comment:
```suggestion
```
Do we need to create the zero arrays? Would just copying the arrays on the following lines suffice?